autoprotocol/protocol.py

```diff
@@ -6521,6 +6527,19 @@ def location_helper(source, destination, volume, method):
             for _ in volume
         ]
 
+        if density:
+            if not isinstance(density, list):
+                density = [density] * count
+            density = [parse_unit(d, "mg/ml") for d in density]
+            for d in density:
+                if d.magnitude <= 0:
+                    raise ValueError(
+                        "Density: {} must be a value larger than 0."
+                        "".format(d)
+                    )
+        else:
+            density = [density] * count
```

Comment:
if the `if density` above (line 6530) evaluates to False, what happens here?

Reply:
Whether density is specified or is None, density will be turned into a list of [density] (line 6540-6541). The `if` statement here checks `density` value is valid if provided. 
... it is not checking for the list length if density is provided in a list, so I'm updating with additional test.